Trove.js

```diff
@@ -0,0 +1,337 @@
+{
+	"translatorID": "8efcb7cb-4180-4555-969a-08e8b34066c4",
+	"label": "Trove",
+	"creator": "Tim Sherratt",
+	"target": "^https?://trove\\.nla\\.gov\\.au/(?:newspaper|gazette|work|book|article|picture|music|map|collection)/",
+	"minVersion": "2.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsbv",
+	"lastUpdated": "2016-04-04 13:06:56"
+}
+
+/*
+   Trove Translator
+   Copyright (C) 2016 Tim Sherratt (tim@discontents.com.au, @wragge)
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU Affero General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Affero General Public License for more details.
+
+   You should have received a copy of the GNU Affero General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+function detectWeb(doc, url) {
+	if (url.indexOf('/result?') != -1 || url.indexOf('/newspaper/page') != -1) {
+		return "multiple";
+	} else if (url.indexOf('/newspaper/article') != -1) {
+		return "newspaperArticle";
+	} else if (url.indexOf('/work/') != -1) {
+		return "document";
+	}
+}
+
+function getSearchResults(doc, url) {
+	var items = {};
+	var results;
+	if (url.indexOf('/result?') != -1) {
+		results = ZU.xpath(doc, "//div[@id='mainresults']//li/dl/dt/a");
+	} else if (url.indexOf('/newspaper/page') != -1) {
+		results = ZU.xpath(doc, "//ol[@class='list-unstyled articles']/li/h4/a");
+	}
+	for (var i=0; i<results.length; i++) {
+		var url = results[i].href;
+		var title = ZU.trimInternal(results[i].textContent);
+		items[url] = title;
+	}
+	return items;
```

Comment:
generally, our pattern has been to return `false` if there were no results and the object otherwise. It's a bit annoying to check that there is nothing in the object otherwise.